generate-docs.sh

```diff
@@ -0,0 +1,57 @@
+#!/bin/sh
+
+set -e
+
+# Build API documentation, Matlab (.m) -> Markdown (.md)
+#   Beware: you MUST put `quit` at the end; Matlab does not auto-quit.
+#   https://www.mathworks.com/matlabcentral/answers/523194-matlab-script-in-batch-from-unix-command-line
+# TODO: broken.
+#  1. writes to temp/shimming-toolbox/docs instead of docs/
+#  2. incompatible with Matlab R2019a which is what's on our CI machine.
+#matlab -nodisplay -nosplash -r "run('genDoc/generate_doc.m');exit"
```

Comment:
**big caveat**: this doesn't actually run matlab yet. Matlab works on this environment, but the code we need to run on it is not compatible with the version ♈ 

Either matlab needs upgrading or the code needs some scalpels taken to it.

By the way, that script has a lot of assumptions. What I would like it to be is:

```
gen_matlab_api_doc(path_to_code_input, path_to_markdown_output)
```

Reply:
as per our slack discussion, Matlab will be upgraded today by @alexfoias